core/src/test/java/com/datastax/oss/driver/api/core/cql/StatementBuilderTest.java

```diff
@@ -16,61 +16,95 @@
+      extends StatementBuilder<MockSimpleStatementBuilder, SimpleStatement> {
 
-    public NullStatementBuilder() {
+    public MockSimpleStatementBuilder() {
       super();
     }
 
-    public NullStatementBuilder(SimpleStatement template) {
+    public MockSimpleStatementBuilder(SimpleStatement template) {
       super(template);
     }
 
     @Override
     public SimpleStatement build() {
-      return null;
+
+      SimpleStatement rv = mock(SimpleStatement.class);
+      when(rv.isTracing()).thenReturn(this.tracing);
+      when(rv.getRoutingKey()).thenReturn(this.routingKey);
+      return rv;
     }
   }
 
   @Test
   public void should_handle_set_tracing_without_args() {
 
-    NullStatementBuilder builder = new NullStatementBuilder();
-    assertThat(builder.tracing).isFalse();
+    MockSimpleStatementBuilder builder = new MockSimpleStatementBuilder();
+    assertThat(builder.build().isTracing()).isFalse();
     builder.setTracing();
-    assertThat(builder.tracing).isTrue();
+    assertThat(builder.build().isTracing()).isTrue();
   }
 
   @Test
   public void should_handle_set_tracing_with_args() {
 
-    NullStatementBuilder builder = new NullStatementBuilder();
-    assertThat(builder.tracing).isFalse();
+    MockSimpleStatementBuilder builder = new MockSimpleStatementBuilder();
+    assertThat(builder.build().isTracing()).isFalse();
     builder.setTracing(true);
-    assertThat(builder.tracing).isTrue();
+    assertThat(builder.build().isTracing()).isTrue();
     builder.setTracing(false);
-    assertThat(builder.tracing).isFalse();
+    assertThat(builder.build().isTracing()).isFalse();
   }
 
   @Test
-  public void should_override_template() {
+  public void should_override_set_tracing_in_template() {
 
     SimpleStatement template = SimpleStatement.builder("select * from system.peers").build();
-    NullStatementBuilder builder = new NullStatementBuilder(template);
-    assertThat(builder.tracing).isFalse();
+    MockSimpleStatementBuilder builder = new MockSimpleStatementBuilder(template);
+    assertThat(builder.build().isTracing()).isFalse();
     builder.setTracing(true);
-    assertThat(builder.tracing).isTrue();
+    assertThat(builder.build().isTracing()).isTrue();
 
     template = SimpleStatement.builder("select * from system.peers").setTracing().build();
-    builder = new NullStatementBuilder(template);
-    assertThat(builder.tracing).isTrue();
+    builder = new MockSimpleStatementBuilder(template);
+    assertThat(builder.build().isTracing()).isTrue();
     builder.setTracing(false);
-    assertThat(builder.tracing).isFalse();
+    assertThat(builder.build().isTracing()).isFalse();
+  }
+
+  private ByteBuffer randomBuffer(Random random) {
+
+    byte[] arr = new byte[8];
+    random.nextBytes(arr);
+    return ByteBuffer.wrap(arr);
+  }
+
+  @Test
+  public void should_match_set_routing_key_vararg() {
+
+    Random random = new Random();
+    ByteBuffer buff1 = randomBuffer(random);
```

Comment:
Do we really need random byte buffers? What about just this:

```
ByteBuffer buff1 = Bytes.fromHexString("0xcafe");
ByteBuffer buff2 = Bytes.fromHexString("0xbabe");
```

Reply:
I don't have a strong opinion either way I guess.  I'm sympathetic to the argument for randomized testing (to potentially catch data-driven oddness).  I'm also sympathetic to the argument for predictable testing to catch regressions.

If there's a preference for a fixed value here I can make that change as well.

Reply:
Yeah I hear you; but I think I have a slight preference for test repeatability. Random tests, unless we are really dealing with randomness, are usually of little value; here for instance, I can't see why the actual byte buffer contents would matter – except, of course, for special values such as `null` or empty.

Reply:
Okay, I'll shift this over to a fixed string in the next round of changes.